jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/go/GoSamDebugSupport.kt

```diff
@@ -15,6 +36,92 @@ class GoSamDebugSupport : SamDebugSupport {
+
+                        if (processHandler == null || processHandler.isStartNotified) {
+                            // this branch will never be hit because it gets here too fast, but it's needed for correctness
+                            process.connect(socketAddress)
+                        } else {
+                            processHandler.addProcessListener(
+                                object : ProcessAdapter() {
+                                    override fun startNotified(event: ProcessEvent) {
+                                        ApplicationThreadPoolScope("GoSamDebugAttacher").launch {
+                                            // If we don't wait, then then debugger will try to attach to
+                                            // the container before it starts Devle. So, we have to add a sleep.
+                                            // Delve takes quite a while to start in the sam cli images hence long sleep
+                                            // See https://youtrack.jetbrains.com/issue/GO-10279
+                                            // TODO revisit this to see if higher IDE versions help FIX_WHEN_MIN_IS_211 (?)
+                                            // 2000 is what JB uses in this scenario, so use it as our default
+                                            delay(Registry.intValue("aws.sam.goDebuggerDelay", 2000).toLong())
+                                            process.connect(socketAddress)
+                                        }
+                                    }
+                                }
+                            )
+                        }
+                        return process
+                    }
+                }
+
+                runInEdt {
+                    promise.setResult(debug)
+                }
+            } catch (t: Throwable) {
+                LOG.warn(t) { "Failed to start debugger" }
+                runInEdt {
+                    promise.setError(t)
+                }
+            }
+        }
+
+        return promise
+    }
+
+    override fun samArguments(runtime: Runtime, packageType: PackageType, debugPorts: List<Int>): List<String> = buildList {
+        val debugger = copyDlv()
+        add("--debugger-path")
+        add(debugger)
+        add("--debug-args")
+        if (packageType == PackageType.IMAGE) {
+            add(
+                "/var/runtime/aws-lambda-go delveAPI -delveAPI=2 -delvePort=${debugPorts.first()} -delvePath=/tmp/lambci_debug_files/dlv"
+            )
+        } else {
+            add("-delveAPI=2")
+        }
+    }
+
+    private fun copyDlv(): String {
+        val dlvFolder = getDelve()
+        val directory = Files.createTempDirectory("goDebugger")
+        Files.copy(dlvFolder.resolve("dlv").toPath(), directory.resolve("dlv"))
+        // Delve that comes packaged with the IDE does not have the executable flag set
+        directory.resolve("dlv").toFile().setExecutable(true)
```

Comment:
Hmm.... What happens if host OS is Windows?

Reply:
Actually this works on windows as is, SAM cli must be doing something funky on windows but it works:
![Capture](https://user-images.githubusercontent.com/11964782/104359692-d3488300-54c4-11eb-84da-56bbd2b52c75.PNG)